pkgs/applications/networking/instant-messengers/signal-desktop/generic.nix

```
@@ -158,7 +158,8 @@ stdenv.mkDerivation rec {
 
     # Fix the desktop link
     substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace "/opt/${dir}/${pname}" $out/bin/${pname}
+      --replace "/opt/${dir}/${pname}" $out/bin/${pname} \
+      --replace "StartupWMClass=Signal" "StartupWMClass=signal"
```

```suggestion
      --replace-fail "StartupWMClass=Signal" "StartupWMClass=signal"
```
--replace is deprecated by now, superceded by -fail/-warn/-quiet variants